dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/Relation.java

```diff
@@ -175,7 +191,7 @@ protected BooleanFormula getExecPair(Event e1, Event e2, SolverContext ctx) {
             e1 = e2;
             e2 = temp;
         }
-        BranchEquivalence eq = task.getBranchEquivalence();
+        BranchEquivalence eq = analysisContext.get(BranchEquivalence.class);
```

Comment:
why not `requires()`?

Reply:
I did not do it because `RelationAnalysis` did this already. And you can think of those methods as being part of `RelationAnalysis` in some extended sense. It's unfortunate that a lot of "external" code is mixed in the relation classes right now but I think that will change once we reworked the analysis.
But for now, I can also change it to `requires` (I guess we can replace all `get` by `requires`, it will cause no harm).